lib/Distribution/Helper/Discover.hs

```diff
@@ -54,6 +56,14 @@ findProjects dir = execWriterT $ do
   let stackYaml = dir </> "stack.yaml"
   whenM (liftIO $ doesFileExist stackYaml) $
     tell [Ex $ ProjLocStackYaml stackYaml]
+
+  cabalVersion <- liftIO $ let ?verbose = (<=0)
```

Comment:
Not sure what behaviour you're trying to get here, but `const False` would probably be more appropriate here.

The function you assign there should always either behave like `withVerbosity` (see below) i.e. `(>=log_level)` or be just `const True` or `const False`.

```
withVerbosity :: (Verbose => IO a) -> IO a
withVerbosity act = do
  x <- lookup  "CABAL_HELPER_DEBUG" <$> getEnvironment
  let ?verbose = \level ->
        case x >>= readMaybe of
          Just x | x >= level -> True
          _ -> False
  act
```

It's a bit ad-hoc I admit :)


There are two definition sites in GhcSession.hs where this is violated, but I think those are bugs and should actually be flipped. It's not like it matters with the mess of output the tests produce ;)